src/store/events/actions.ts

```diff
@@ -1,13 +1,48 @@
-import {Event, GET_EVENTS, EventActionTypes} from './types'
-import {bindActionCreators, Dispatch, AnyAction} from 'redux';
-import * as mocks from '../../mocks';
-import {ThunkAction} from 'redux-thunk';
+import {Event} from '../../models'
+import {GET_EVENTS, EventActionTypes, ADD_EVENTS} from './types'
+import {error as Error} from '../types'
+import {bindActionCreators, Dispatch, AnyAction, ActionCreator} from 'redux';
+import {GetData} from '../../services';
+import {ThunkAction, ThunkDispatch} from 'redux-thunk';
+import realm from '../../realm';
 
 export const getEvents = (): ThunkAction<EventActionTypes, {}, {}, AnyAction> => (
   dispatch: Dispatch,
 ): EventActionTypes => {
+  const events = realm.objects<Event>("Event").map(realmObject => ({
+    ...realmObject
+  }))
   return dispatch({
     type: GET_EVENTS,
-    events: mocks.events,
+    events: events,
   });
-};
\ No newline at end of file
+};
+
+export const loadEvents : ActionCreator<
+  ThunkAction<Promise<EventActionTypes>, {}, void, AnyAction>
+> = () => {
+  return async (dispatch: ThunkDispatch<{}, {}, any>): Promise<EventActionTypes> => {
+    let events : Event[] = []
+    try {
+      events = await GetData.Events();
+      // const text = await Api.call();
+    } catch (error) {
+      return dispatch({
+        type : Error.REQUEST_ERROR,
+        error
+      })
+    }
+    realm.write(() => {
```

Comment:
What is this for? From what I can tell, you are deleting all events if they exist and then readding them to the db. I can understand initially to populate the db, but the performance of this is kinda yikes after initially populated.

Reply:
Yeah this is jsut a temporrary poplating thing I jsut wanted to see if it would work with retrieving data from realm we aren't actually gonna do this for the realm integration